New translation: write-pages.md (#228)
gatsbyjs/gatsby-pt-BR: docs/docs/write-pages.md

---
title: Escrever páginas
---

> Esta documentação não está atualizada com a versão mais recente do Gatsby.
>
> As áreas que estão desatualizadas são:
>
> - `data.json` deve ser substituído por `page-data.json`
> - remover menções de `pages.json`
> - descrever `match-paths.json`
>

> > Você pode ajudar fazendo um PR para [update this documentation](https://github.com/gatsbyjs/gatsby/issues/14228).

Reviewer: ```suggestion
> Você pode ajudar fazendo um PR para [atualize esta documentação](https://github.com/gatsbyjs/gatsby/issues/14228).
```

Reviewer: Vi que usou alguns links em inglês outros em português, então vou marcar para deixar em português.

> Este é um dos últimos estágios de bootstrap antes de enviarmos para o webpack para executar a otimização e divisão do código. O Webpack cria um pacote para a web. Ele não tem conhecimento do código principal da Gatsby. Em vez disso, ele opera em arquivos no diretório `.cache`. Ele também não tem acesso a todas as informações do Redux que foram criadas durante a inicialização. Em vez disso, criamos arquivos Javascript e JSON dinâmicos que são dependentes da aplicação do webpack no diretório `.cache` (veja [Building the JavaScript App](/docs/production-app/)).

Reviewer: ```suggestion
Este é um dos últimos estágios de bootstrap antes de enviarmos para o webpack para executar a otimização e divisão do código. O Webpack cria um pacote para a web. Ele não tem conhecimento do código principal da Gatsby. Em vez disso, ele opera em arquivos no diretório `.cache`. Ele também não tem acesso a todas as informações do Redux que foram criadas durante a inicialização. Em vez disso, criamos arquivos Javascript e JSON dinâmicos que são dependentes da aplicação do webpack no diretório `.cache` (veja [Construindo o aplicativo JavaScript](/docs/production-app/)).
```

## pages.json

Esta é uma coleção de objetos de página, criados a partir do namespace redux `pages`. Para cada página, inclui o

- [componentePedacoNome](/docs/behind-the-scenes-terminology/#componentchunkname)
- [jsonNome](/docs/behind-the-scenes-terminology/#jsonname)
- [caminho](/docs/behind-the-scenes-terminology/#path)
- [combineCaminho](/docs/behind-the-scenes-terminology/#matchpath)

> As páginas são classificadas de forma que aquelas com `matchPath` venham antes daquelas sem. Isso é para ajudar.

Reviewer 2: Acho que está parte ta duplicada

Author: Boa, tá mesmo!

## pages.json

Esta é uma coleção de objetos de página, criados a partir do namespace redux `pages`. Para cada página, inclui o

> - [componentePedacoNome](/docs/behind-the-scenes-terminology/#componentchunkname)

Reviewer: ```suggestion
- [componentePedacoNome](/docs/behind-the-scenes-terminology/#componentchunkname)
```

> - [componentePedacoNome](/docs/behind-the-scenes-terminology/#componentchunkname)
> - [jsonNome](/docs/behind-the-scenes-terminology/#jsonname)
> - [caminho](/docs/behind-the-scenes-terminology/#path)
> - [combineCaminho](/docs/behind-the-scenes-terminology/#matchpath)

Reviewer: ```suggestion
- [componentChunkName](/docs/behind-the-scenes-terminology/#componentchunkname)
- [jsonName](/docs/behind-the-scenes-terminology/#jsonname)
- [path](/docs/behind-the-scenes-terminology/#path)
- [matchPath](/docs/behind-the-scenes-terminology/#matchpath)
```

Reviewer: Creio que neste caso é melhor manter o nome original pois quando o usuário for utilizar a ferramenta ele encontrará esses nomes [dentro do código](https://www.gatsbyjs.org/docs/gatsby-internals-terminology/#redux-components-namespace).

## async-requires.js

---

O `async-requires.js` é muito semelhante ao `sync-requires.js`, pois é um arquivo JavaScript gerado dinamicamente. A diferença é que ele foi escrito para ser usado na divisão de código via webpack. Portanto, ao invés de usar `require` com o caminho do componente, ele usa `import` e adiciona uma dica `webpackChunkName` para que possamos eventualmente vincular o componentChunkName ao arquivo resultante (mais informações em [Code Splitting](/docs/how-code-splitting-works/) docs). `components` é uma função, para que possa ser inicializada lentamente.

`async-requires.js` também exporta uma função `data` que importa `data.json` ([see below](/docs/write-pages/#datajson))

Um exemplo de requisição assíncrona é:

```javascript
exports.components = {
  "component---src-blog-2-js": () =>
    import(
      "/home/site/src/blog/2.js" /* webpackChunkName: "component---src-blog-2-js" */
    ),
  // mais componentes
};

exports.data = () => import("/home/site/.cache/data.json");
```

> Lembre-se de que, `sync-requires.js` é usado durante [Page HTML Generation](/docs/html-generation/). E o `async-requires.js` é usado por [Building the JavaScript App](/docs/production-app/).

Reviewer: ```suggestion
Lembre-se de que, `sync-requires.js` é usado durante [Geração de HTML da página](/docs/html-generation/). E o `async-requires.js` é usado por [Construindo o aplicativo JavaScript](/docs/production-app/).
```

## data.json

> Este é um arquivo json gerado. Ele contém todo o conteúdo do `pages.json` ([como acima](/docs/write-pages/#pagesjson)), e todo o redux `jsonDataPaths` que foi criado no final do [Execução de Consulta](/docs/query-execution/#save-query-results-to-redux-and-disk) estágio. Então, isso parece com:

Reviewer: ```suggestion
Este é um arquivo json gerado. Ele contém todo o conteúdo do `pages.json` ([como acima](/docs/write-pages/#pagesjson)), e todo o redux `jsonDataPaths` que foi criado no final da [Execução de Consulta](/docs/query-execution/#save-query-results-to-redux-and-disk) estágio. Então, isso parece com:
```

Author: Então, aqui eu usei `do` porque eu acredito que a referência seja para a palavra `estágio` que vem depois dos colchetes e parênteses. Você acha melhor referenciar ao que vem dentro do colchetes?

Reviewer 3: Dentro dos colchetes, pois vai virar texto corrido, os colchetes são do markdown para adicionar links.

Reviewer 3: Sugiro:
`que foi criado no final da [execução de consulta](/docs/query-execution/#save-query-results-to-redux-and-disk) do estágio.`

que resultara em:
que foi criado no final da [execução de consulta](/docs/query-execution/#save-query-results-to-redux-and-disk) do estágio.

Author: Boa!

        "jsonName": "blog-2-c06",
        "path": "/blog/2"
    },
    // mais componentes
 ],

 // jsonName -> dataPath
 dataPaths: {
   "blog-2-c06":"952/path---blog-2-c06-meTS6Okzenz0aDEeI6epU4DPJuE",
   // mais componentes
 }
```

`data.json` é usado em dois lugares. Primeiro, ele é importado lentamente por `async-requires.js` (acima), que por sua vez é usado por `production-app` para [carregar resultados do json](/docs/production-app/#load-page-resources) para uma página.

Também é usado pela [Geração de HTML da página](/docs/html-generation/) de duas maneiras:

1. `static-entry.js` produz um pacote da web `page-renderer.js` agrupado pelo webpack que gera o HTML para um caminho. Isto requer `data.json` e usa as `pages` para procurar página por página.

> 2. Para obter `jsonName` do objeto da página e usá-lo para construir um caminho de rescurso para o resultado real do json, procurando em `data.json.dataPaths[jsonName]`.

Reviewer: ```suggestion
2. Para obter `jsonName` do objeto da página e usá-lo para construir um caminho de recurso para o resultado real do json, procurando em `data.json.dataPaths[jsonName]`.
```
